terraform/backend-state/dynamodb.tf

```diff
@@ -0,0 +1,14 @@
+resource "aws_dynamodb_table" "terraform_statelock" {
```

Comment:
Where do you store the remote state, which created the remote state bucket? ;)

Reply:
as per readme.. If you want to use remote state. then you have to run the `backend-state` content first if you want it to work out the box. again i looked at complex bool approach to turn on / off remote state usage.. but couldnt figure out a way of creating a logical switch inside the `terraform {}` definition

Reply:
will now make the backend-state and optional module call and then variablise the bucket and dynao name etc so that user can use their own existing backend if they want